Add more details to OIDC guides (#45929)
quarkusio/quarkus: docs/src/main/asciidoc/security-oidc-code-flow-authentication.adoc

== Using the authorization code flow mechanism

=== Configuring Quarkus to support authorization code flow

To enable an authorization code flow authentication, the `quarkus.oidc.application-type` property must be set to `web-app`.

> Usually, the Quarkus OIDC ``web-app`` application type must be set when your Quarkus application is a frontend application which serves HTML pages and requires an OIDC single sign-on login.

Reviewer: ```suggestion
Usually, the Quarkus OIDC `web-app` application type must be set when your Quarkus application is a frontend application which serves HTML pages and requires an OIDC single sign-on login.
```

I added double quotes since the suggestion was in quotes, sorry...

== Using the authorization code flow mechanism

> === Configuring the Quarkus application to redirect the user

Reviewer: It would be useful to have this section indeed, we describe in the quickstart but sure, let's have the one in the ref doc as well. But I'd like to I propose to rework it a little bit, to avoid focusing specifically on the redirect...

How about having a section title `Enabling authorization code flow` or may be `Configuring Quarkus to support authorization code flow` ? The 2nd option is probably better as the follow up one also starts with `Configuring...`

> As outlined in the previous section, the Authorization Code Flow begins with the user being redirected to the OIDC provider for authentication.

Reviewer: ```suggestion
```

> To enable this redirect, the `quarkus.oidc.application-type` property must be set to `web-app`. This tells Quarkus that the

Reviewer: ```suggestion
To enable an authorization code flow, the `quarkus.oidc.application-type` property must be set to `web-app`.
```

> application serves pages (usually a front-end application).

Reviewer: Quarkus security does not really know what the application does, may be we can indeed add some clarification, I'll try to comment more tomorrow
```suggestion
```

Reviewer: How about, `Usually, the Quarkus OIDC `web-app` application type must be set when your Quarkus application is a frontend application which serves HTML pages.` ? Or something like that

> For this type of client, the Authorization Code Flow is defined as the preferred method for authenticating users.

Reviewer: The client is indeed a correct OIDC term but it is quite technical IMHO,

```suggestion
For the Quarkus OIDC `web-app`application, the authorization code flow is defined as the preferred method for authenticating users.
```

> For applications that serve both a front-end and a REST API, the `hybrid` setting can be used instead. In this case, the Authentication

Reviewer: ```suggestion
For applications that support both the authorization code flow and bearer access token authentication, the `hybrid` application type can be used instead.
```
